nobl9/resource_slo.go

```diff
@@ -1102,6 +1103,23 @@ func schemaMetricCloudwatch() map[string]*schema.Schema {
 				"(https://docs.nobl9.com/Sources/Amazon_CloudWatch/#creating-slos-with-cloudwatch)",
 			Elem: &schema.Resource{
 				Schema: map[string]*schema.Schema{
+					"account_id": {
+						Type:     schema.TypeString,
+						Optional: true,
+						ValidateDiagFunc: func(i interface{}, path cty.Path) diag.Diagnostics {
+							value := i.(string)
+							var diags diag.Diagnostics
+							if m, err := regexp.MatchString(`^[0-9]{12}$`, value); !m || err != nil {
+								diags = append(diags, diag.Diagnostic{
+									Severity: diag.Error,
+									Summary:  "account_id must be 12-digit identifier",
+								})
+							}
+
+							return diags
+						},
```

Comment:
Why not use this?

```suggestion
			ValidateDiagFunc: validation.ToDiagFunc(
				validation.IntBetween(12, 12),
			),
```

Reply:
account_id have to be 12 digit number, `validation.IntBetween(12, 12)` checks if int value is between 12 and 12 :D

Reply:
omg, potato brain here xD nvm then